testutil/src/main/java/org/spine3/test/NullToleranceTest.java

```diff
@@ -0,0 +1,500 @@
+        }
+
+        private boolean validateException(InvocationTargetException ex) {
+            final Throwable cause = ex.getCause();
+            checkException(cause);
+            final boolean result = hasExpectedStackTrace(cause);
+            return result;
+        }
+
+        private static void checkException(Throwable cause) {
+            final boolean correctException = cause instanceof NullPointerException;
+            if (!correctException) {
+                throw Exceptions.wrappedCause(cause);
+            }
+        }
+
+        /**
+         * Checks the stack trace elements.
+         *
+         * <p> It is expected that each of the tested utility methods invokes
+         * {@link Preconditions#checkNotNull(Object)} as a first step of the execution.
+         *
+         * <p> Therefore the stack trace is analysed to ensure its first element references
+         * the {@code Preconditions#checkNotNull(Object)} and the second one references the tested method.
+         *
+         * @param cause the {@code Throwable}
+         * @return {@code true} if the {@code StackTraceElement}s matches the expected, {@code false} otherwise
+         */
+        private boolean hasExpectedStackTrace(Throwable cause) {
+            final StackTraceElement[] stackTraceElements = cause.getStackTrace();
+            final StackTraceElement preconditionsElement = stackTraceElements[0];
+            final boolean preconditionClass = Preconditions.class.getName()
+                                                                 .equals(preconditionsElement.getClassName());
+            if (!preconditionClass) {
+                return false;
+            }
+
+            final StackTraceElement targetClassElement = stackTraceElements[1];
+            final String targetMethodName = targetClassElement.getMethodName();
+            final boolean correct = method.getName()
+                                          .equals(targetMethodName);
+            if (!correct) {
+                return false;
+            }
+
+            final boolean correctClass = targetClassName.equals(targetClassElement.getClassName());
+            return correctClass;
+        }
+    }
+
+    /**
+     * Provides the default values for the method arguments.
+     */
+    private static class DefaultValuesProvider {
+
+        private static final Class[] EMPTY_PARAMETER_TYPES = {};
+        private static final Object[] EMPTY_ARGUMENTS = {};
+        private static final String STRING_DEFAULT_VALUE = "";
+        private static final String METHOD_NAME = "getDefaultInstance";
+        private final Map<?, ?> defaultValues;
+
+        private DefaultValuesProvider(Map<?, ?> defaultValues) {
+            this.defaultValues = defaultValues;
+        }
+
+        /**
+         * Returns the default value for the method argument by the method argument type.
+         * If default value does not provide, throws {@link IllegalStateException} otherwise.
```

Comment:
```
(blank line)
<p>If a default value for the given type is not customized, an {@code IllegalStateException} is thrown.
```
And let's rename the method argument variable. `key` is too specific to the implementation.

Also, you should probably describe the behaviour with the parent-child types (custom default value types vs. the `key`).